### PR TITLE
E2e deploy demo contract and required changes to cctld

### PR DESCRIPTION
**Pull request by marijanp**
- [ ]

### CODE REVIEW

**Review by Avi-D-coder — APPROVED** [review]

**Review by Avi-D-coder — APPROVED** [review]

File: nixos/tests/end-to-end.nix [hunk 1]
```diff
@@ -31,69 +32,79 @@ nixosTest {
...
-    assert "ok\n" in cli_output
-
-    cli_output = client.succeed("kairos-cli withdraw --amount 1000 --private-key ${testResources}/secp256k1/secret_key.pem")
-    assert "ok\n" in cli_output
-  '';
+  testScript =
+    let
+      cctlUsersDirectory = "/var/lib/cctl/assets/users";
+      # port see: https://github.com/casper-network/cctl/blob/745155d080934c409d98266f912b8fd2b7e28a00/utils/constants.sh#L13
+      casperNodeAddress = "http://localhost:11101";
```
Comment (koxu1996) [inline nixos/tests/end-to-end.nix:39]: **Comment:** I was surprised to see port `11100` at above URL, but I found that [CCTL uses base port concatenated with node ID](https://github.com/casper-network/cctl/blob/ec014c13781d779738922dd1dd283435b04a1740/utils/infra.sh#L211) (starting at 1).

File: nixos/tests/end-to-end.nix [hunk 2]
```diff
@@ -31,69 +32,79 @@ nixosTest {
...
+      kairos.wait_for_unit("kairos.service")
+      kairos.wait_for_unit("nginx.service")
+      kairos.wait_for_open_port(80)
+
+      client.wait_for_unit ("multi-user.target")
+
+      kairos.succeed("casper-client get-node-status --node-address ${casperNodeAddress}")
+
+      # chain-name see: https://github.com/casper-network/cctl/blob/745155d080934c409d98266f912b8fd2b7e28a00/utils/constants.sh#L66
+      kairos.succeed("casper-client put-deploy --node-address ${casperNodeAddress} --chain-name cspr-dev-cctl  --secret-key $(ls ${cctlUsersDirectory}/user-1/secret_key.pem) --payment-amount 5000000000000  --session-path ${kairos-contracts}/bin/demo-contract-optimized.wasm")
```
Comment (koxu1996) [inline nixos/tests/end-to-end.nix:57]: Removed extra spaces:

```suggestion
      kairos.succeed("casper-client put-deploy --node-address ${casperNodeAddress} --chain-name cspr-dev-cctl --secret-key $(ls ${cctlUsersDirectory}/user-1/secret_key.pem) --payment-amount 5000000000000 --session-path ${kairos-contracts}/bin/demo-contract-optimized.wasm")
```

Comment (koxu1996) [reply nixos/tests/end-to-end.nix:57]: **Suggestion:** `cspr-dev-cctl` could be replaced with top-level variable, e.g. `casperChainName`.

File: kairos-test-utils/src/cctl.rs [hunk 3]
```diff
@@ -30,15 +30,20 @@ pub struct CasperNode {
 }
 
 pub struct CCTLNetwork {
-    pub working_dir: TempDir,
-    pub assets_dir: PathBuf,
```
Comment (koxu1996) [inline kairos-test-utils/src/cctl.rs:34]: **Comment:** Assets directory path was used in `drop()`, and now we are basing on assumption it's placed directly in the workdir and named `assets`. That is okay though.

**Review by koxu1996 — APPROVED** [review]

**Comment by marijanp** [comment]
I will fixup the whitespaces in a successive PR, I want to avoid dismissing the reviews. Ideally the chain name would be parametrizable by cctl, which is not the case currently.

**Pull request merged**